Client/game_sa/CPedSA.cpp

```diff
@@ -565,389 +376,210 @@ CVector* CPedSA::GetTransformedBonePosition(eBone bone, CVector* vecPosition)
 }
 
-bool CPedSA::IsBleeding()
+std::uint32_t CPedSA::GetFootBlood()
 {
-    return GetPedInterface()->pedFlags.bPedIsBleeding;
+    CPedSAInterface* pedInterface = GetPedInterface();
+    
+    // If the foot blood flag is activated, return the amount of foot blood
+    return pedInterface->pedFlags.bDoBloodyFootprints ? pedInterface->dwTimeWhenDead : 0;
 }
 
-void CPedSA::SetBleeding(bool bBleeding)
+void CPedSA::SetOnFire(bool onFire)
 {
-    GetPedInterface()->pedFlags.bPedIsBleeding = bBleeding;
-}
+    CPedSAInterface* pedInterface = GetPedInterface();
 
-bool CPedSA::IsOnFire()
-{
-    if (GetPedInterface()->pFireOnPed != NULL)
-        return true;
-    return false;
-}
+    // If we are already on fire, don't apply a new fire
+    if ((onFire && pedInterface->pFireOnPed) || (!onFire && !pedInterface->pFireOnPed))
+    {
+        return;    
+    }
 
-void CPedSA::SetOnFire(bool bOnFire)
-{
-    CPedSAInterface* pInterface = GetPedInterface();
+    CFireManagerSA* fireManager = static_cast<CFireManagerSA*>(pGame->GetFireManager());
+    if (!fireManager)
+        return;
 
-    if (bOnFire)
+    if (onFire)
     {
-        // If we are already on fire, don't apply a new fire
-        if (pInterface->pFireOnPed == NULL)
-        {
-            CFireManagerSA* pFireManager = static_cast<CFireManagerSA*>(pGame->GetFireManager());
-            CFire*          pFire = pFireManager->StartFire(this, NULL, (float)DEFAULT_FIRE_PARTICLE_SIZE);
+        CFire* fire = fireManager->StartFire(this, nullptr, static_cast<float>(DEFAULT_FIRE_PARTICLE_SIZE));
+        if (!fire)
+            return;
 
-            if (pFire)
-            {
-                // Start the fire
-                pFire->SetTarget(this);
-                pFire->Ignite();
-                pFire->SetStrength(1.0f);
-                // Attach the fire only to the player, do not let it
-                // create child fires when moving.
-                pFire->SetNumGenerationsAllowed(0);
-                pInterface->pFireOnPed = pFire->GetInterface();
-            }
-        }
+        fire->SetTarget(this);
+        fire->Ignite();
+        fire->SetStrength(1.0f);
+
+        // Attach the fire only to the player, do not let it
+        // create child fires when moving.
+        fire->SetNumGenerationsAllowed(0);
+        pedInterface->pFireOnPed = fire->GetInterface();
     }
     else
     {
-        // Make sure that we have some attached fire
-        if (pInterface->pFireOnPed != NULL)
-        {
-            CFireManagerSA* pFireManager = static_cast<CFireManagerSA*>(pGame->GetFireManager());
-            CFire*          pFire = pFireManager->GetFire(static_cast<CFireSAInterface*>(pInterface->pFireOnPed));
-
-            if (pFire)
-            {
-                pFire->Extinguish();
-            }
-        }
+        CFire* fire = fireManager->GetFire(pedInterface->pFireOnPed);
+        if (fire)
+            fire->Extinguish();
```

Comment:
early return

Reply:
?

Reply:
You can use early return here to increase readability

Reply:
I don't understand what's going on. Maybe github shows something wrong? Early return is already used here

```cpp
void CPedSA::SetOnFire(bool onFire)
{
    CPedSAInterface* pedInterface = GetPedInterface();

    // If we are already on fire, don't apply a new fire
    if ((onFire && pedInterface->pFireOnPed) || (!onFire && !pedInterface->pFireOnPed))
        return;    

    CFireManagerSA* fireManager = static_cast<CFireManagerSA*>(pGame->GetFireManager());
    if (!fireManager)
        return;

    if (onFire)
    {
        CFire* fire = fireManager->StartFire(this, nullptr, static_cast<float>(DEFAULT_FIRE_PARTICLE_SIZE));
        if (!fire)
            return;

        fire->SetTarget(this);
        fire->Ignite();
        fire->SetStrength(1.0f);

        // Attach the fire only to the player, do not let it
        // create child fires when moving.
        fire->SetNumGenerationsAllowed(0);
        pedInterface->pFireOnPed = fire->GetInterface();
    }
    else
    {
        CFire* fire = fireManager->GetFire(pedInterface->pFireOnPed);
        if (fire)
            fire->Extinguish();
    }
}
```

Reply:
Oh well, yeah. Github didnt show me the changes. Nevermind it then